sklearnex/preview/covariance/covariance.py

```diff
@@ -100,4 +106,27 @@ def fit(self, X, y=None):
 
         return self
 
+    # This needs to be included to guarantee that we use the sklearnex pairwise_distances
+    @wrap_output_data
+    def mahalanobis(self, X, queue=None):
+        if sklearn_check_version("1.0"):
+            X = self._validate_data(X, reset=False)
+        else:
+            X = check_array(X)
+
+        precision = self.get_precision()
+        with config_context(assume_finite=True):
+            # compute mahalanobis distances
+            dist = pairwise_distances(
+                X, self.location_[np.newaxis, :], metric="mahalanobis", VI=precision
+            )
+
+        return np.reshape(dist, (len(X),)) ** 2
+
+    error_norm = wrap_output_data(sklearn_EmpiricalCovariance.error_norm)
+    score = wrap_output_data(sklearn_EmpiricalCovariance.score)
```

Comment:
It make sense generally have a mechanism, that wraps output data for all non-patched method of the estimator.

Reply:
Agreed, this should be a follow up PR which we will have to add to the backlog (I imagine it will be a long undertaking)